### DISCUSSION TITLE
Using an external USB DAC as audio sink

### DISCUSSION THREAD

**Discussion by yubiuser** [pschatzmann/arduino-audio-tools#1830] [Q&A] [answered]
I have an ESP32-WROVER Dev Kit and want to use it for an little audio project. I have a speaker with a USB port acting as a DAC itself when connected to a PC. 
Is it possible to connect the ESP32 via it's USB to the USB input of the speaker? Which audio sink would I need to configure in `audio-tools` to output via USB?

**Comment by pschatzmann** [comment, marked as answer]
No basic ESP32 does not support USB and I have never seen any Arduino examples that implement an Audio USB host on processors that support USB.

**Reply by yubiuser** [reply]
Thanks. Then I'll use the TOSLINK input :-)

**Comment by yubiuser** [comment]
Oh...

https://docs.espressif.com/projects/esp-iot-solution/en/latest/usb/usb_overview/usb_host_solutions.html#esp-usb-audio-solution

**Comment by pschatzmann** [comment]
You still have the issue that this does not support your ESP32-WROVER
